### ISSUE TITLE
DKSY 

### ISSUE THREAD

**Issue by Alexustas** [Alexustas/Props-requests#20]
![screenshot336](https://cloud.githubusercontent.com/assets/6713671/10876420/e5e5086c-8166-11e5-98c9-865020805c53.png)

https://youtu.be/EFRlWGmAOLQ

**Comment by MOARdV** [comment]
Did I mention that I'm starting to work on a new IVA for the FASA Apollo capsule?  This prop looks like it provides a lot of the functionality that I've cobbled together in the Gemini IVA between the various numeric displays.

**Comment by Alexustas** [comment]
I still can not decide whether to make a second, independent, copy of the device

**Comment by MOARdV** [comment]
What do you mean by a second independent copy?  A -02 version that can display different data?  I think, for most IVA, a single unit would be sufficient.

**Comment by Alexustas** [comment]
```
   A -02 version that can display different data? 
```

Yes, exactly

**Comment by MOARdV** [comment]
I don't know if a second one is needed.  As a heads-up, I am thinking of adding another launch timer variable - something like TARGETPLANETIMESECS to provide a countdown timer for launching into the plane of a target.

**Comment by Alexustas** [comment]
DL: https://www.dropbox.com/s/98wcambq8vncsny/ASET_DSKY.zip?dl=0

**Comment by MOARdV** [comment]
I've got some feedback on this instrument.  I've spent the last week on spaceplanes, so it took me longer than planned to place this in an IVA.

I have not figured out what the MNVR source (top row button) does.  every time I use it, the top three lines are '---------' (even when there is a maneuver node).  Is there supposed to be data displayed in that mode?

When Source = orbit, if I press delta = MNVR, I see -999999 instead of ----------- (pretty sure it was Source = orbit - maybe it was one of the other source settings).

I see ATTITUDE ERROR light up fairly regularly when maneuver in orbit.  I am not sure why.  It does not stay on a long time.

One thing that I think is missing is a way to display current altitude - maybe when Source = Orbit, there could be a button that can display current altitude instead of inclination (this could be valuable for TRGT mode, too, so you can see the target's current altitude)?  I know there's an analog altimeter, but I've edited my altimeter so it doesn't work when ALTITUDE > ORBITBODYATMOSPHERETOP (it's a static pressure altimeter :) ).

**Comment by Alexustas** [comment]
> I have not figured out what the MNVR source (top row button) does

Source 'MNVR' - it's for future use. i hope sooner or later we will get an IVA maneuver editor 
(You can regard it as an official request :) ), so MNVR will display the expected parameters of the orbit after the maneuver (Apoapsis, Periapsis anad Inclination)

**Comment by Alexustas** [comment]
> I see ATTITUDE ERROR light up fairly regularly when maneuver in orbit.

This lamp illuminates when the deviation "Error Needles" exceeds a certain value. I made specially for the case where the pilot looks only at DSKY and does not see FDAI. The sensitivity of this indicator dependent on the selected scale of "Error Needles"

**Comment by Alexustas** [comment]
> way to display current altitude

I thought about it and came to the conclusion that in the space flight, current altitude is not so important. we have a value of Apoapsis and Periapsis and time remaining to them, in most cases this is enough

**Comment by Alexustas** [comment]
> When Source = orbit, if I press delta = MNVR, I see -999999 instead of ----------- (pretty sure it was Source = orbit - maybe it was one of the other source settings).

I'll check that.
DELTA MNVR should show Delta V remaining in the current maneuver node (if exist)

**Comment by MOARdV** [comment]
> DELTA MNVR should show Delta V remaining in the current maneuver node (if exist)

Yes, I see that, and TIMER MNVR also works, but SOURCE MNVR doesn't seem to do anything.  Although I see you said above why, so disregard.

**Comment by MOARdV** [comment]
> When Source = orbit, if I press delta = MNVR, I see -999999 instead of ----------- (pretty sure it was Source = orbit - maybe it was one of the other source settings).

I was wrong here - it's Source = RNDZ that shows -999999 for Delta = MNVR when there is no node.

**Comment by MOARdV** [comment]
Not DSKY problem, but I think the FDAI GMCP Radial + / - is backwards.  I think Radial In should use

```
      SOURCE_VARIABLE
      {
        name = PERSISTENT_FDAIGMCPDirectionPlus-01
        range = -0.1, 0.1
      }
```

and Radial Out uses opposite switch position.  I notice this when testing the SAS and MechJeb controls in a cockpit. When MechJeb was set to Radial +, FDAI had to be set to Radial - to show the same heading.

**Comment by MOARdV** [comment]
I've started on documentation for this prop (it took me long enough):

https://github.com/MOARdV/FlightSystemsRedux/wiki/How-to-DSKY

**Comment by Alexustas** [comment]
Oh! really great job!

**Comment by MOARdV** [comment]
Question on DSKY: SCD_ScreenLine04Obj has text 'TIME', but I do not see a transform to illuminate the block behind it, so the text is not easy to see.

SCD_ScreenLine05Obj has 'dV' text, but same thing - it is not illuminated.

Are there transforms to add lights behind these texts?

**Comment by Alexustas** [comment]
I completely forgot about these things, because they are not interactive, like the rest of the lights ((

transform for "TIME" light - **TIMELightsObj**
for "DV" - **DVLightsObj**

these lights glow when electricity is available on board

**Comment by MOARdV** [comment]
> I completely forgot about these things, because they are not interactive, like the rest of the lights ((

Too many lights. ))  

I think I will have the DSKY working in MAS this week.  I think it will be much easier to understand how it works looking at a Lua script file instead of many, many CUSTOM and SELECT variables.

**Comment by Alexustas** [comment]
![2016-08-24 00-39-55-484](https://cloud.githubusercontent.com/assets/6713671/17905024/64068616-6993-11e6-960a-f7eef49ef587.jpg)

**Comment by Alexustas** [comment]
> Lua script file instead of many, many CUSTOM and SELECT variables.

yes, Lua script gets much smaller and easier readable. 
After a few hours of studying Lua, I made this sketch of the script for "tuning" navigation radio 

```
function setRadioFreq(freq)
    StationFound,StationIsNDB,StationIsVOR,StationIsILS = false,false,false,false
    CurrentStationName,CurrentStationLat,CurrentStationLong,CurrentStationType,CurrentStationAlt = 0,0,0,0,0
    local count = 0
    for _ in pairs(StationFreq) do count = count + 1 end
    for i= 1, count do
       if StationFreq[i]==freq then 
            StationFound = true
            CurrentStationName,CurrentStationLat,CurrentStationLong,CurrentStationType,CurrentStationAlt = StationName[i],StationLat[i],StationLong[i],StationType[i],StationAlt[i] 
            if StationType[i] == "NDB" then StationIsNDB = true end
            if StationType[i] == "VOR" then StationIsVOR = true end
            if StationType[i] == "ILS" then StationIsILS = true end
       end
    end
end 
```

but even I have no idea how to implement reading definitions of stations from config. So now this function uses such tables:

```
local StationFreq =
{
    108.1,
    110.1,
    113.3,
    113.5,
    114.2,
    114.6,
    115.1,
    115.2,
    116.2,
    116.4,
    117.3
}

local StationName =
{
    "AAA1",
    "AAA2",
    "AAA3",
    "AAA4",
    "AAA5",
    "AAA6",
    "AAA7",
    "AAA8",
    "AAA9",
    "AA10",
    "AA11"
}

local StationLat =
{
    -0.050235,
    -0.050235,
    -0.050235,
    -0.050235,
    -0.048742,
    -0.048519,
    -0.317261,
    -0.096786,
    -0.071619,
    -1.516030,
    -1.848429
}

local StationLong =
{
    -74.63320,
    -71.91117,
    -61.26513,
    -62.7539,
    -37.0459,
    -99.7559,
    -136.5601,
    -120.3223,
    -151.4575,
    -19.2485,
    -21.7969
}

local StationType =
{
    "VOR",
    "ILS",
    "NDB",
    "NDB",
    "VOR",
    "VOR",
    "NDB",
    "NDB",
    "VOR",
    "VOR",
    "VOR"
}

local StationAlt =
{
    115,
    254,
    1182,
    652,
    247,
    587,
    691,
    1357,
    2051,
    1436,
    875
}
```

**Comment by MOARdV** [comment]
> but even I have no idea how to implement reading definitions of stations from config.

MAS will support NAV stations, so the script will be able to query MAS.  I have some ideas on how to include the information in a config file, since I think KSP 1.2 waypoints can be named.

```
NAV_WAYPOINT (I don't know what this name will be)
{
  name = VOR:AAA1@108.1
  position = -0.050235, -74.63320
  altitude = 115 // or -1 to say "At ground level"?
  body = Kerbin // maybe I want a beacon at my Mun base :)
}
```

As long as KSP allows me to read the waypoint database, any new beacon can be added using that name format, and MAS will be able to say, "A-ha!  This waypoint is a VOR named AAA1, and its frequency is 108.1".  And even if I can not read the database, there can still be config files that define waypoints (so they may be added automatically to the database).

And then I just need to know what things the props will need to query (direction, distance, ?).
